src/main/java/net/fabricmc/loader/FabricLoader.java

```diff
@@ -153,16 +153,21 @@ public void load() {
 				modText = "Loading %d mods";
 				break;
 			case 1:
-				modText = "Loading %d mod: %s";
+				modText = "Loading %d mod:";
 				break;
 			default:
-				modText = "Loading %d mods: %s";
+				modText = "Loading %d mods:";
 				break;
 		}
 
-		LOGGER.info("[" + getClass().getSimpleName() + "] " + modText, candidateMap.values().size(), candidateMap.values().stream()
-			.map(info -> String.format("%s@%s", info.getInfo().getId(), info.getInfo().getVersion().getFriendlyString()))
-			.collect(Collectors.joining(", ")));
+		LOGGER.info("[" + getClass().getSimpleName() + "] " + modText, candidateMap.values().size());
+
+		List<ModCandidate> candidates = new ArrayList<>(candidateMap.values());
+		candidates.sort(Comparator.comparing(candidate -> candidate.getInfo().getId()));
+
+		for (ModCandidate candidate : candidates) {
+			LOGGER.info("- %s@%s", candidate.getInfo().getId(), candidate.getInfo().getVersion().getFriendlyString());
```

Comment:
I think chloe's concern is valid. Don't think loggers respect java format

Reply:
Fabric Loader uses `getFormatterLogger` instead of `getLogger`, which uses the `String.format` formatting instead of standard Log4J formatting.